tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/HealthCheck.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tool.profiling
+
+import com.nvidia.spark.rapids.tool.ToolTextFileWriter
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.sql.rapids.tool.profiling.ApplicationInfo
+
+/**
+ * HealthCheck defined health check rules
+ */
+class HealthCheck(apps: ArrayBuffer[ApplicationInfo], textFileWriter:ToolTextFileWriter){
+
+  require(apps.nonEmpty)
+
+  // Function to list all failed tasks , stages and jobs.
+  def listFailedJobsStagesTasks(): Unit = {
+    for (app <- apps) {
+      // Look for failed tasks.
+      val tasksMessageHeader = s"Application ${app.appId} (index=${app.index}) failed tasks:\n"
```

Comment:
I think the rest of the headers leave off the appid and app index, we should be consistent about that.